crates/sparrow-backend/src/logical_to_physical.rs

```diff
@@ -50,7 +50,12 @@ pub(super) struct LogicalToPhysical {
 #[derive(Debug)]
 struct Reference {
     /// Reference to a specific step.
-    step_id: sparrow_physical::StepId,
+    ///
+    /// This should only be `None` if the expression is a literal. It should
+    /// not be `None` if the expression is a literal arising within a specific
+    /// step (domain) -- only for cases where the value originally has no
+    /// domain.
+    step_id: Option<sparrow_physical::StepId>,
```

Comment:
During compilation, literals don't belong to any step. This actually helps some with some of the weird literal behavior.

1. When a literal is first converted, we say `step_id=None, expr=[CodeForLiteral]`.
2. When that is used (merged) with other arguments, we add the expression to the merged step (which may be a specific step if all arguments are in the same step or following a merge if there is one).

But at that point, we have a reference to that literal in a specific step.